File: Visual Basic/Factor Calculator 4.vb
Module Module1

    Sub Main()
        Console.WriteLine("Please input your number.")
        Dim i As Long = Console.ReadLine()
        Console.Write(Environment.NewLine)
        If (i > 0) Then
            For d = 1 To Math.Sqrt(i)
                If (i Mod d = 0) Then
                    Console.WriteLine("{0}, {1}", i / d, d)
                End If
            Next
            Console.Write(Environment.NewLine)
            Console.WriteLine("The list of factors has been successfully generated.")
        Else
            Console.WriteLine("What are you doing?")
        End If
    End Sub

End Module

File: C/Factor Calculator 4.c
#include <stdio.h>
#include <math.h>

void main() {
    unsigned long long n, d;
    printf("What is your number?\n");
    scanf("%llu", &n);
    printf("\n");
    if (n > 0) {
        for (d = 1; d <= sqrt(n); d++)
            if (n % d == 0)
                printf("%llu, %llu\n", n / d , d);
        printf("\nThe list of factor pairs has been successfully generated.\n");
    }
    else
        printf("\nInvalid number!\n");
}
